### PR TITLE
Using spring amqp bean to setup tls

### PR DESCRIPTION
**Pull request by sandeep-sen**
- [x] I have read the [CONTRIBUTION GUIDE][contributing]

- [ ] This is a bug

- [x] This is a feature request

- [ ] This is a syntax/style fix

- [x] This is an improvement to code quality

---
**Please describe your changes below:**

_"Changed the implementation for SSL connection to AMQP."_
_"It uses spring-rabbit `RabbitConnectionFactoryBean` ."_

**Usage example**

[Link to example in credential-service-parent (refactor-wip branch)](https://eos2git.cec.lab.emc.com/VCE-Symphony/credential-service-parent/blob/refactor-wip/credential-service/src/main/java/com/dell/cpsd/credential/amqp/config/ProductionConfig.java#L51)

```java
@Bean
@Qualifier("rabbitConnectionFactory")
    public ConnectionFactory productionCachingConnectionFactory()
    {
        RabbitMQCachingConnectionFactory cachingCF = null;
        com.rabbitmq.client.ConnectionFactory connectionFactory;
        try{
            if (propertiesConfig.isSslEnabled())
            {
                RabbitMQTLSFactoryBean rabbitMQTLSFactoryBean = new RabbitMQTLSFactoryBean(propertiesConfig);
                connectionFactory = rabbitMQTLSFactoryBean.getObject();
            }
            else
            {
                connectionFactory = new com.rabbitmq.client.ConnectionFactory();
            }
            cachingCF = new RabbitMQCachingConnectionFactory(connectionFactory, propertiesConfig);
        }
        catch (Exception e){
            e.printStackTrace();
        }
        return cachingCF;
    }
```

[contributing]: http://dellemc-symphony.readthedocs.io/en/latest/contributingtosymphony.html

### CODE REVIEW

File: common-contract-enricher/pom.xml [hunk 1]
```diff
@@ -9,7 +9,7 @@
     <parent>
         <groupId>com.dell.cpsd</groupId>
         <artifactId>common-messaging-parent</artifactId>
-        <version>2.0.1-SNAPSHOT</version>
+        <version>2.1.1-SNAPSHOT</version>
```
Comment (bonczj) [inline common-contract-enricher/pom.xml]: Has another team already used 2.1.0-SNAPSHOT? Not a big deal, but looks a bit odd only changing the minor version.

File: common-rabbitmq/src/main/java/com/dell/cpsd/common/rabbitmq/config/IRabbitMqPropertiesConfig.java [hunk 2]
```diff
@@ -108,31 +108,39 @@
      * the property is <code>trustStorePassphrase</code>
      *
      * @return The trustStorePassphrase
+     * @Deprecated
```
Comment (bonczj) [inline common-rabbitmq/src/main/java/com/dell/cpsd/common/rabbitmq/config/IRabbitMqPropertiesConfig.java]: What is the expectation around removing the Deprecated code? I understand a transition period is needed, but at the same time, we need to ensure that time period is still time bound.

File: common-rabbitmq/src/main/java/com/dell/cpsd/common/rabbitmq/connectors/RabbitMQTLSFactoryBean.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+ * @since SINCE-TDB
+ */
+public class RabbitMQTLSFactoryBean extends RabbitConnectionFactoryBean {
+
+    private static final ILogger LOGGER = RabbitMQLoggingManager.getLogger(RabbitMQTLSFactoryBean.class);
+
+    public RabbitMQTLSFactoryBean(IRabbitMqPropertiesConfig configuration) {
+        super();
+        try{
+            String trustStorePath = System.getProperty("javax.net.ssl.trustStore");
```
Comment (bonczj) [inline common-rabbitmq/src/main/java/com/dell/cpsd/common/rabbitmq/connectors/RabbitMQTLSFactoryBean.java]: With the changes to the configuration (above) and pulling in these system properties instead, I expect to see the README.md updated to explain how to correctly connect to RabbitMQ. That will help the teams understand the change without having to pull you in every time. :)

File: common-rabbitmq/src/main/java/com/dell/cpsd/common/rabbitmq/connectors/RabbitMQTLSFactoryBean.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+            this.setUseSSL(configuration.isSslEnabled());
+            this.setSslAlgorithm(configuration.tlsVersion());
+            this.setKeyStoreType("JKS");
+            this.setTrustStoreType("JKS");
+            this.setTrustStoreResource(new FileSystemResource(trustStorePath));
+            this.setTrustStorePassphrase(trustStorePassphrase);
+            this.setKeyStoreResource(new FileSystemResource(trustStorePath));
+            this.setKeyStorePassphrase(trustStorePassphrase);
+            this.afterPropertiesSet();
```
Comment (bonczj) [inline common-rabbitmq/src/main/java/com/dell/cpsd/common/rabbitmq/connectors/RabbitMQTLSFactoryBean.java]: Was there any input on key strength from PSO? It should be validated in here as a minimum. I suspect they asked for 2048-bit minimum with a configuration option to ensure 4096-bit.

Comment (sandeep-sen) [reply common-rabbitmq/src/main/java/com/dell/cpsd/common/rabbitmq/connectors/RabbitMQTLSFactoryBean.java]: The key is coming from vault, so it is 2048-bit minimum.

File: common-rabbitmq/src/main/java/com/dell/cpsd/common/rabbitmq/config/IRabbitMqPropertiesConfig.java (hunk 2 above)
Comment (sandeep-sen) [reply common-rabbitmq/src/main/java/com/dell/cpsd/common/rabbitmq/config/IRabbitMqPropertiesConfig.java]: I was told whenever we bump up the major version we will get rid of the deprecated code, unfortunately no time limit was specified :( 👎

File: common-rabbitmq/src/main/java/com/dell/cpsd/common/rabbitmq/connectors/RabbitMQTLSFactoryBean.java (hunk 3 above)
Comment (sandeep-sen) [reply common-rabbitmq/src/main/java/com/dell/cpsd/common/rabbitmq/connectors/RabbitMQTLSFactoryBean.java]: Updated readme

**Comment by bonczj** [comment]
Thanks Sandeep for the updates!

Can we look at defaulting RabbitMQPropertiesConfig.java to be secure by default? Set a default tlsVersion of 1.2 and sslEnabled=TRUE. Right now, sslEnabled defaults to false and tlsVersion is left empty.

File: common-rabbitmq/src/main/java/com/dell/cpsd/common/rabbitmq/config/IRabbitMqPropertiesConfig.java (hunk 2 above)
Comment (giddyelysium) [reply common-rabbitmq/src/main/java/com/dell/cpsd/common/rabbitmq/config/IRabbitMqPropertiesConfig.java]: Ideally go to the minor version with @deprecated, once you have released the minor version, immediately go to next major and release that removing the code. So instant depreciation (ensure communication with any teams that have branches or forks is key)

File: common-rabbitmq/src/main/java/com/dell/cpsd/common/rabbitmq/connectors/RabbitMQTLSFactoryBean.java (hunk 4 above)
Comment (sandeep-sen) [reply common-rabbitmq/src/main/java/com/dell/cpsd/common/rabbitmq/connectors/RabbitMQTLSFactoryBean.java]: Link to Language specific extensions story
https://jira.cec.lab.emc.com:8443/browse/ESTS-133282

Will be completed by this week and will update readme also.

**Pull request merged**